crates/sui-graphql-rpc/src/context_data/pg_backend.rs

```diff
@@ -631,16 +635,41 @@ impl PgQueryExecutor for PgManager {
         self.inner
             .spawn_blocking(move |this| {
                 let query = query_builder_fn()?;
-                let explain_result: String = this
+                let explain_result: Option<String> = this
                     .run_query(|conn| query.explain().get_result(conn))
-                    .map_err(|e| Error::Internal(e.to_string()))?;
-                let cost = extract_cost(&explain_result)?;
-                if cost > max_db_query_cost as f64 {
-                    return Err(DbValidationError::QueryCostExceeded(
-                        cost as u64,
-                        max_db_query_cost,
-                    )
-                    .into());
+                    .tap_err(|e| {
+                        warn!(
+                            target: EXPLAIN_COSTING_LOG_TARGET,
+                            "Failed to get explain result: {}", e
+                        )
+                    })
+                    .ok(); // Fine to not propagate this error as explain-based costing is not critical today
+
+                if let Some(explain_result) = explain_result {
+                    let cost = extract_cost(&explain_result)
+                        .tap_err(|e| {
+                            warn!(
+                                target: EXPLAIN_COSTING_LOG_TARGET,
+                                "Failed to get cost from explain result: {}", e
+                            )
+                        })
+                        .ok(); // Fine to not propagate this error as explain-based costing is not critical today
+
+                    if let Some(cost) = cost {
+                        if cost > max_db_query_cost as f64 {
+                            warn!(
+                                target: EXPLAIN_COSTING_LOG_TARGET,
+                                cost,
+                                max_db_query_cost,
+                                exceeds = true
+                            );
+                        } else {
+                            info!(
+                                target: EXPLAIN_COSTING_LOG_TARGET,
+                                cost,
+                            );
+                        }
+                    }
                 }
```

Comment:
Oh I see, well you still don't need to use `ok`, you can use `let ... else`, to avoid the nested `if let`:

```rust
let Ok(explain_result) = this
    .run_query(|conn| query.explain().get_result(conn))
    .tap_err(|e| {
        warn!(
            target: EXPLAIN_COST_LOG_TARGET,
            "Failed to get explain result: {e}",
       )
    })
else { /* Consume error */ };

let Ok(cost) = extract_cost(&explain_result)
    .tap_err(|e| {
        warn!(
            target: EXPLAIN_COSTING_LOG_TARGET,
            "Failed to get cost from explain result: {e}",
        )
    })
else { /* Consume error */ };

if cost > max_db_query as f64 {
    warn!(
        target: EXPLAIN_COSTING_LOG_TARGET,
        cost,
        max_db_query_cost,
        exceeds = true
    );
} else {
    info!(
        target: EXPLAIN_COSTING_LOG_TARGET,
        cost,
    )
}
```

Also my 2c is that defining a constant for your log target was probably overkill -- the constant itself is probably shorter to type than the variable name, it's only used in this file, so it can easily be "find and replaced" if you needed to, and the variable's contents are probably not very different from its name.